src/components/Shop.vue

```diff
@@ -40,7 +40,11 @@ export default {
     },
   },
   created() {
-    this.$store.dispatch('requestMachine');
+    const payload = {
+      idmodule: this.$route.query.idmodule,
+      moduleNameToinstall: this.$route.query.module_name_toinstall,
+    };
+    this.$store.dispatch('requestMachine', payload);
```

Comment:
It works, but I'd like to improve this by removing automatically these details from the URL as soon as they've been taken in account.